src/ui/events.js

```diff
@@ -25,38 +27,141 @@ export type MapMouseEvent = {
         | 'mouseleave'
         | 'mouseover'
         | 'mouseout'
-        | 'contextmenu',
-    map: Map,
-    originalEvent: MouseEvent,
-    point: Point,
-    lngLat: LngLat
-};
+        | 'contextmenu';
+
+    /**
+     * The `Map` object that fired the event.
+     */
+    map: Map;
+
+    /**
+     * The DOM event which caused the map event.
+     */
+    originalEvent: MouseEvent;
+
+    /**
+     * The pixel coordinates of the mouse cursor, relative to the map and measured from the top left corner.
+     */
+    point: Point;
+
+    /**
+     * The geographic location on the map of the mouse cursor.
+     */
+    lngLat: LngLat;
+
+    /**
+     * Prevents subsequent default processing of the event by the map.
+     *
+     * Calling this method will prevent the following default map behaviors:
+     *
+     *   * On `mousedown` events, the behavior of {@link DragPanHandler}
+     *   * On `mousedown` events, the behavior of {@link DragRotateHandler}
+     *   * On `mousedown` events, the behavior of {@link BoxZoomHandler}
```

Comment:
I think we should also have:
 - On `wheel` events, the behavior of `{@link ScrollZoomHandler}`

Reply:
#6225